src/+datools/+statistical/+ensemble/EnF.m

```diff
@@ -52,34 +54,29 @@
             addParameter(p, 'RIPIterations', 0);
             addParameter(p, 'RankHistogram', [])
             addParameter(p, 'ResamplingThreshold', 0.5);
+            addParameter(p, 'EnsembleGenerator', @(x, y) randn(x, y));
+            
             parse(p, varargin{:});
 
             s = p.Results;
+            
+            obj@datools.DABase(p.Unmatched);
 
-            obj.Model = s.Model;
-            obj.ModelError = s.ModelError;
             obj.Inflation = s.Inflation;
             obj.Rejuvenation = s.Rejuvenation;
             obj.Localization = s.Localization;
             obj.Parallel = s.Parallel;
             obj.RankHistogram = s.RankHistogram;
             obj.ResamplingThreshold = s.ResamplingThreshold;
             ensN = s.NumEnsemble;
+            
             if ~isempty(obj.RankHistogram)
                 RankValue = zeros(length(obj.RankHistogram), ensN+2);
             end
 
-            kept = p.Unmatched;
-
-            p = inputParser;
-            addParameter(p, 'Observation', datools.observation.Observation(s.Model.NumVars));
-            addParameter(p, 'EnsembleGenerator', @(x) randn(s.Model.NumVars, x));
-            parse(p, kept);
-
-            s = p.Results;
-
             obj.Ensemble = s.EnsembleGenerator(ensN);
-            obj.Observation = s.Observation;
+            obj.Anomalies = (1/sqrt(ensN)) * (obj.Ensemble - mean(obj.Ensemble, 2));
```

Comment:
Anomalies need to be a dependent variable

Reply:
I want anomalies and mean to be independent variables. That way, anytime we inflate, we don't need to compute the mean, compute the anomalies, and inflate, and recreate the ensemble. We just inflate the anomalies and recreate the ensemble.

Since we want to use obj.inflate() inside the filtering methods as well, a lot of computations are redone for inflation and then the filtering. Keeping them independent lets the filtering method have direct access to mean, anomalies. Reduces a ton of redundant computation.

Yes, it takes more storage, which increases linearly by a factor of 2.

Reply:
Computing mean and anomalies is super super cheap.

>We just inflate the anomalies and recreate the ensemble.

So we halve the cost.

In turn we use more storage AND the interface becomes A LOT more convoluted, as just changing the anomalies will NOT recompute the ensemble. That is super unintuitive.